### DISCUSSION TITLE
Multiple subscriptions for 1 channel

### DISCUSSION THREAD

**Discussion by techwizzdom** [supabase/supabase#18130] [Questions] [answered]
I'm building a notification system in my React/Supabase app, and in App.tsx, I create a subscription:

```
const subscription = supabase
      .channel('schema-db-changes')
      .on(
        'postgres_changes',
        {
          event: 'INSERT',
          schema: 'public',
          table: 'chat_message_notifications',
          filter: `recipient_id=eq.${currentUser?.id}`,
        },
        handleNewChatMessageNotification
      )
      .subscribe()
```

And it works perfectly.

This subscription should be active on every page - however, I have another one for the chat that should be active only if a chat thread is opened:

```
const subscription = supabase
      .channel('schema-db-changes')
      .on(
        'postgres_changes',
        {
          event: 'INSERT',
          schema: 'public',
          table: 'chat_messages',
          filter: `chat_thread_id=eq.${chatThreadId}`,
        },
        handleNewChatMessage
      .subscribe()
```

And this 2nd subscription destroys the 1st one, once I open a chat thread. I "fixed" it by listening to `'table-db-changes'` channel, but that doesn't seem like a long term solution - what should I do when I'll have to listen to subscriptions in multiple places in the app?

**Comment by GaryAustin1** [comment, marked as answer]
Change the name of the channel between the two...

**Reply by techwizzdom** [reply]
Do I also need to define those channels somewhere, or can I just do `.channel('random-name')`?

**Reply by GaryAustin1** [reply]
Yes just make up different names.

**Comment by dquang-tran** [comment]
So is that correct to conclude that supabase doesn't support multiple subscriptions with the same channel name per client?
